ingress/controllers/nginx-third-party/controller.go

```diff
@@ -329,6 +220,113 @@ func (lbc *loadBalancerController) registerHandlers() {
 	glog.Fatalf(fmt.Sprintf("%v", http.ListenAndServe(fmt.Sprintf(":%v", *healthzPort), nil)))
 }
 
+func (lbc *loadBalancerController) sync() {
+	// defaut/some-ingress -> default-some-ingress
+	// name := strings.Replace(key, "/", "-", -1)
+
+	ings := lbc.ingLister.Store.List()
+
+	lbc.updateNGINX(ing)
+
+	lbc.nginx.Reload()
+}
+
+func (lbc *loadBalancerController) updateNGINX(ings []extensions.Ingress) {
+	pems := make(map[string]string)
+
+	upstreams := make(map[string]nginx.Upstream)
+	// var upstreams []nginx.Upstream
+
+	for _, rule := range ing.Spec.Rules {
+		if rule.IngressRuleValue.HTTP == nil {
+			continue
+		}
+
+		for _, path := range rule.HTTP.Paths {
+			name := ing.Name + "-" + path.Backend.ServiceName
+			ups := nginx.NewUpstreamWithDefaultServer(name)
+
+			svcKey := ing.Namespace + "/" + path.Backend.ServiceName
+			svcObj, svcExists, err := lbc.svcLister.Store.GetByKey(svcKey)
+			if err != nil {
+				glog.Infof("error getting service %v from the cache: %v", svcKey, err)
+			} else {
+				if svcExists {
+					svc := svcObj.(*api.Service)
+					if svc.Spec.ClusterIP != "None" && svc.Spec.ClusterIP != "" {
+						upsServer := nginx.UpstreamServer{Address: svc.Spec.ClusterIP, Port: path.Backend.ServicePort.String()}
+						ups.Backends = []nginx.UpstreamServer{upsServer}
+					} else if svc.Spec.ClusterIP == "None" {
+						endps, err := lbc.endpLister.GetServiceEndpoints(svc)
+						if err != nil {
+							glog.Infof("error getting endpoints for service %v from the cache: %v", svc, err)
+						} else {
+							upsServers := endpointsToUpstreamServers(endps, path.Backend.ServicePort.IntValue())
+							if len(upsServers) > 0 {
+								ups.Backends = upsServers
+							}
+						}
+					}
+				}
+			}
+
+			upstreams = append(upstreams, ups)
+		}
+	}
+
+	var servers []nginx.Server
+	for _, rule := range ing.Spec.Rules {
+		server := nginx.Server{Name: rule.Host}
+
+		if pemFile, ok := pems[rule.Host]; ok {
+			server.SSL = true
+			server.SSLCertificate = pemFile
+			server.SSLCertificateKey = pemFile
+		}
+
+		var locations []nginx.Location
+
+		for _, path := range rule.HTTP.Paths {
+			loc := nginx.Location{Path: path.Path}
+			upsName := ing.GetName() + "-" + path.Backend.ServiceName
+
+			for _, ups := range upstreams {
+				if upsName == ups.Name {
+					loc.Upstream = ups
+				}
+			}
+			locations = append(locations, loc)
+		}
+
+		server.Locations = locations
+		servers = append(servers, server)
+	}
+
+	// lbc.nginx.AddOrUpdateIngress(name, nginx.IngressNGINXConfig{Upstreams: upstreams, Servers: servers})
```

Comment:
are these todos, or something you plan to / have removed in a subsequent commit?